### ISSUE TITLE
'Unexpected end of JSON input' with custom apiBaseUrl

### ISSUE THREAD

**Issue by lvii** [Christopher-Hayes/vscode-chatgpt-reborn#27]
### Describe the Bug

Hi

I have replaced default `chatgpt.gpt3.apiBaseUrl` with the proxy API URL in `settings.json`

```
"chatgpt.gpt3.apiBaseUrl": "https://openai.1rmb.tk/api/v1",
```

return error message : `Unexpected end of JSON input`

I have tested the proxy API with `curl` command it works ok :

```
❯ curl -sS https://chatai.1rmb.tk/api/v1/chat/completions
{
    "error": {
        "message": "You didn't provide an API key. You need to provide your API key in an Authorization header using Bearer auth (i.e. Authorization: Bearer YOUR_KEY), or as the password field (with blank username) if you're accessing the API from your browser and are prompted for a username and password. You can obtain an API key from https://platform.openai.com/account/api-keys.",
        "type": "invalid_request_error",
        "param": null,
        "code": null
    }
}
```

### Where are you running VSCode? (Optional)

None

### Which OpenAI model are you using? (Optional)

None

**Comment by Christopher-Hayes** [comment]
Hi @lvii, can you try using `https://openai.1rmb.tk/v1/` as the API url instead?

**Comment by Christopher-Hayes** [comment]
But in the meantime, I'll push a patch to make the errors a little more helpful and include a specific note about this proxy server for anyone trying to use a path different from /v1.

![image](https://user-images.githubusercontent.com/6013871/230701015-abf7b78c-dc37-4b17-b6ea-ca6bc7177c2d.png)

**Comment by Christopher-Hayes** [comment]
Okay, so I finally figured out what it was -

1. The proxy you use is [open-source](https://github.com/x-dr/chatgptProxyAPI), so I figured I'd just host my own proxy if that one is getting too much traffic.
2. But, my proxy was getting the same "unexpected end of JSON input".
3. It turns out when using a proxy, the first data response from the proxy in some cases will get split in two, which the code wasn't expecting. I'll be pushing a new release that handles, so https://openai.1rmb.tk/v1/ should work for you again.

On a sidenote since I already set it up to debug this, I'll be offering another proxy at https://openai-proxy.dev which runs the same code as openai.1rmb.tk.
